sound/soc/sof/nocodec.c

```diff
@@ -55,35 +38,58 @@ static int nocodec_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
-/* we just set some BEs - FE provided by topology */
-static struct snd_soc_dai_link sof_nocodec_dais[] = {
-	/* Back End DAI links */
-	{
-		/* SSP0 - Codec */
-		.name = "NoCodec",
-		.id = 0,
-		.init = nocodec_rtd_init,
-		.cpu_dai_name = "sof-nocodec-dai",
-		.platform_name = "sof-audio",
-		.no_pcm = 1,
-		.codec_dai_name = "snd-soc-dummy-dai",
-		.codec_name = "snd-soc-dummy",
-		.ops = &sof_nocodec_ops,
-		.dai_fmt = SND_SOC_DAIFMT_DSP_B | SND_SOC_DAIFMT_NB_NF |
-			SND_SOC_DAIFMT_CBS_CFS,
-		.ignore_suspend = 1,
-		.be_hw_params_fixup = sof_nocodec_codec_fixup,
-		.dpcm_playback = 1,
-		.dpcm_capture = 1,
-	},
-};
-
 static struct snd_soc_card sof_nocodec_card = {
 	.name = "sof-nocodec",
-	.dai_link = sof_nocodec_dais,
-	.num_links = ARRAY_SIZE(sof_nocodec_dais),
 };
 
+int sof_nocodec_setup(struct device *dev,
```

Comment:
This file can be completely removed now and this function should be in sof/utils.c
I would also change this function to sof_platform_add_all_bes()

Reply:
They are fine, they should register the card and get around any deferred probe binding.